src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -421,6 +421,36 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId
+     * @param zone
+     * @param instanceId
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public Operation.Error createSnapshot(String projectId, String zone, String instanceId) throws IOException, InterruptedException{
+        //TODO: cleanup in the case of a failed snapshot creation (try catch?)
```

Comment:
These TODOs should be annotated with user github username.

Reply:
These are going to be gone by next PR/CR